racksdb/web/ui/src/views/DatacenterRoomView.vue

```diff
@@ -7,20 +7,48 @@ SPDX-License-Identifier: GPL-3.0-or-later -->
 <script setup lang="ts">
 import { useHttp } from '@/plugins/http'
 import { useRacksDBAPI } from '@/composables/RacksDBAPI'
-import { ref, onMounted } from 'vue'
-import SearchBar from '@/components/SearchBar.vue'
+import { ref, onMounted, computed } from 'vue'
 import type { Ref } from 'vue'
-import type { Datacenter, Infrastructure, Rack } from '@/composables/RacksDBAPI'
+import type { Datacenter, Infrastructure } from '@/composables/RacksDBAPI'
+import { Switch, SwitchGroup, SwitchLabel } from '@headlessui/vue'
+import { BarsArrowDownIcon, BarsArrowUpIcon } from '@heroicons/vue/24/outline'
+import { Dialog, DialogPanel } from '@headlessui/vue'
 
 const http = useHttp()
 const racksDBAPI = useRacksDBAPI(http)
 const datacenters: Ref<Array<Datacenter>> = ref([])
```

Comment:
Do you still really need `datacenters` and `datacenterDetails` references here? It seems only `infrastructures` and `rackDetails` are really required.

Reply:
No `datacenters` and `datacenterDetails` were not needed, so I deleted `datacenters`, `datacenterDetails` is now declared in the function `getDatacenters()`.
Also using `ref` on the variable `infrastructures` wasn't necessary, so i removed the `ref`.